test/distributed/org/apache/cassandra/distributed/test/streaming/BoundExceptionTest.java

```diff
@@ -0,0 +1,144 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.distributed.test.streaming;
+
+import java.io.IOException;
+import java.util.HashSet;
+
+import com.google.common.base.Throwables;
+import org.junit.Test;
+
+import org.apache.cassandra.streaming.StreamSession;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class BoundExceptionTest
+{
+    private static final int LIMIT = 2;
+
+    @Test
+    public void testSingleException() {
+        try {
+            throw new RuntimeException("test exception");
+        } catch (Exception e) {
+            StreamSession.boundStackTrace(e, LIMIT, new HashSet<>());
+            String expectedStackTrace = "java.lang.RuntimeException: test exception\n" +
+                                        "\tat org.apache.cassandra.distributed.test.streaming.BoundExceptionTest.testSingleException(BoundExceptionTest.java:38)\n" +
+                                        "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n";
+            assertEquals(expectedStackTrace,Throwables.getStackTraceAsString(e));
+            assertEquals(e.getStackTrace().length, LIMIT);
+        }
+    }
+
+    @Test
+    public void testNestedException() {
+        try {
+            throw new RuntimeException(new IllegalArgumentException("the disk /foo/var is bad", new IOException("Bad disk somewhere")));
+        } catch (Exception e) {
+            StreamSession.boundStackTrace(e, LIMIT, new HashSet<>());
+            String expectedStackTrace = "java.lang.RuntimeException: java.lang.IllegalArgumentException: the disk /foo/var is bad\n" +
+                                        "\tat org.apache.cassandra.distributed.test.streaming.BoundExceptionTest.testNestedException(BoundExceptionTest.java:52)\n" +
+                                        "\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n" +
+                                        "Caused by: java.lang.IllegalArgumentException: the disk /foo/var is bad\n" +
+                                        "\t... " + LIMIT + " more\n" +
+                                        "Caused by: java.io.IOException: Bad disk somewhere\n" +
+                                        "\t... " + LIMIT + " more\n";
```

Comment:
are you aware what this is saying?  This feels like you copy/pasted the output and added a test showing they match... `"\t... " + LIMIT + " more\n"` shouldn't be present